notify.py

```diff
@@ -4,25 +4,36 @@
 import re
 
 class Notification(object):
+    """ Allows you to register multiple regex patterns with a person
+        This is designed to let you notify users based on an incoming pattern
+    """
 
     def __init__(self):
         self.patterns = {}
 
-    def add_pattern(self, person: str, pattern: str):
+    def add_pattern(self, person: str, pattern: str) -> None:
+        """ register a pattern to notify a given person
+        """
+        
         if person not in self.patterns:
             self.patterns[person] = []
 
         self.patterns[person].append(pattern)
```

Comment:
:art: could do a `re.compile(pattern)` here and notify the user if it fails, even without a message example to match (can be outside this PR)